app/scripts/modules/core/src/managed/overview/artifact/Constraints.tsx

```diff
@@ -0,0 +1,127 @@
+import cx from 'classnames';
+import { isEmpty } from 'lodash';
+import React from 'react';
+
+import { Icon } from '@spinnaker/presentation';
+import { useApplicationContextSafe } from 'core/presentation';
+import { NotifierService } from 'core/widgets';
+
+import { VersionOperationIcon } from './VersionOperation';
+import { constraintsManager } from '../../constraints/registry';
+import { FetchApplicationDocument, useUpdateConstraintMutation } from '../../graphql/graphql-sdk';
+import spinner from '../loadingIndicator.svg';
+import { ArtifactVersionProps, QueryConstraint } from '../types';
+
+import './Constraints.less';
+
+const ConstraintContent = ({
+  constraint,
+  versionProps,
+}: {
+  constraint: QueryConstraint;
+  versionProps: ArtifactVersionProps;
+}) => {
+  const description = constraintsManager.renderDescription(constraint);
+  const actions = constraintsManager.getActions(constraint)?.sort((action) => (action.pass ? -1 : 1)); // positive actions first
+  const application = useApplicationContextSafe();
+
+  const [updateConstraint, { loading, error }] = useUpdateConstraintMutation({
+    refetchQueries: [{ query: FetchApplicationDocument, variables: { appName: application?.name } }],
+  });
+
+  React.useEffect(() => {
+    if (error) {
+      NotifierService.publish({
+        action: 'create',
+        key: 'updateConstraintError',
+        content: `Failed to update constraint - ${error.message}`,
+        options: { type: 'error' },
+      });
+    }
+  }, [error]);
+
+  return (
+    <dl className="constraint-content">
+      {description && <dd>{description}</dd>}
+      {!isEmpty(actions) && (
+        <dd>
+          {actions?.map(({ title, pass }) => (
+            <button
+              className={cx('btn md-btn constraint-action-button', pass ? 'md-btn-success' : 'md-btn-danger')}
+              key={title}
+              disabled={loading}
+              onClick={() => {
+                updateConstraint({
+                  variables: {
+                    application: application.name,
+                    environment: versionProps.environment,
+                    status: {
+                      artifactVersion: versionProps.version,
+                      type: constraint.type,
+                      artifactReference: versionProps.reference,
+                      status: pass ? 'FORCE_PASS' : 'FAIL',
+                    },
+                  },
+                });
+              }}
+            >
+              {title}
+            </button>
+          ))}
+          {loading && <img src={spinner} height={14} />}
+        </dd>
+      )}
+    </dl>
+  );
+};
+
+const Constraint = ({
```

Comment:
It's really painful to use `CollapsibleSection` as I need to override so many of the styles, while it's pretty simple to create this component from scratch

Reply:
You've probably detected a theme in my review comments: I want to consolidate things together instead of making more snowflakes that cause both the code and the UX to diverge further.  It's worth investing in reusing existing components.  If they aren't usable as-is, let's figure out how to get them up to par.

Reply:
Let's talk about it, it might be easier to come up with a proper solution together